src/Buckle/Buckle/CodeAnalysis/Binding/Binder.cs

```diff
@@ -1142,17 +1171,21 @@ private struct BinderState {
         var operandType = operandTemp.typeClause;
 
         var tempOp = BoundUnaryOperator.Bind(expression.op.type, operandType);
+        var tempDiagnostics = new BelteDiagnosticQueue();
 
         if (tempOp == null)
-            diagnostics.Push(
+            tempDiagnostics.Push(
                 Error.InvalidUnaryOperatorUse(expression.op.location, expression.op.text, operandType));
 
         EndEmulation(binderSaveState);
 
+        // catch casting errors before they happen
+        diagnostics.Move(tempDiagnostics);
+
         if (diagnostics.FilterOut(DiagnosticType.Warning).Any())
             return new BoundErrorExpression();
 
-        var opType = tempOp.typeClause;
+        var opType = tempOp?.typeClause;
```

Comment:
I added this nullable check operator for debugging and left it in as it could theoretically prevent some null exceptions